btor/src/boolector_solver.cpp

```diff
@@ -624,8 +634,22 @@ Term BoolectorSolver::make_symbol(const std::string name, const Sort & sort)
   return term;
 }
 
+Term BoolectorSolver::make_param(const std::string name, const Sort & sort)
+{
+  std::shared_ptr<BoolectorSortBase> bs =
+      std::static_pointer_cast<BoolectorSortBase>(sort);
+  BoolectorNode * n = boolector_param(btor, bs->sort, name.c_str());
+  Term term = std::make_shared<BoolectorTerm>(btor, n);
+  return term;
+}
+
 Term BoolectorSolver::make_term(Op op, const Term & t) const
 {
+  if (op.prim_op == Forall || op.prim_op == Exists)
```

Comment:
i don't understand this check. Can you explain what you are doing here?

Reply:
It's not a valid use of `make_term` with those operators, because it needs to take exactly two terms: the parameter to be bound, and the body term. Since that function only takes an `op` and one `term`, it can't be a valid use. So the exception is saying, you need to use a parameter and a body term with a quantifier op. Does that make sense?